### PR TITLE
[mono] Implement generics GetNodesInGroup

### PR DESCRIPTION
**Pull request by paulloz**
Talked about this (godotengine/godot-proposals#1059) a little bit with neikeq over on Discord.  

This isn't at all an urgent matter. Feel free to tell me if anything's wrong, I'm not 100% sure about my knowledge on this and I'll be happy to improve whatever should be improved.

✌️

### CODE REVIEW

File: modules/mono/glue/scene_tree_glue.cpp [hunk 1]
```diff
@@ -0,0 +1,81 @@
+/*************************************************************************/
+/*  collections_glue.cpp                                                 */
```
Comment (neikeq) [inline modules/mono/glue/scene_tree_glue.cpp:2]:
```suggestion
/*  scene_tree_glue.cpp                                                 */
```

File: modules/mono/glue/scene_tree_glue.cpp [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        if (klass == GDMonoUtils::get_class_native_base(klass)) {
+            // If we're trying to get native objects, just check the inheritance list
+            StringName native_class_name = GDMonoUtils::get_native_godot_class_name(klass);
+            for (int i = 0; i < nodes.size(); ++i)
+                if (ClassDB::is_parent_class(nodes[i]->get_class(), native_class_name))
+                    ret.push_back(nodes[i]);
+        } else {
+            // If we're trying to get csharpscript instances, get the mono object and compare the classes
+            for (int i = 0; i < nodes.size(); ++i) {
+                auto si = CAST_CSHARP_INSTANCE(nodes[i]->get_script_instance());
```
Comment (neikeq) [inline modules/mono/glue/scene_tree_glue.cpp:62]: Code style forbids `auto`.
```suggestion
                CSharpInstance *si = CAST_CSHARP_INSTANCE(nodes[i]->get_script_instance());
```

File: modules/mono/glue/scene_tree_glue.cpp [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+                if (si != NULL) {
```
Comment (neikeq) [inline modules/mono/glue/scene_tree_glue.cpp:64]: Code style enforces `nullptr` over `NULL`.
```suggestion
                if (si != nullptr) {
```

File: modules/mono/glue/scene_tree_glue.cpp [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+                    MonoObject *obj = si->get_mono_object();
+                    if (obj != NULL && mono_object_get_class(obj) == mono_class)
```
Comment (neikeq) [inline modules/mono/glue/scene_tree_glue.cpp:66]: ditto
```suggestion
                    if (obj != nullptr && mono_object_get_class(obj) == mono_class)
```

File: modules/mono/glue/scene_tree_glue.cpp [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+        MonoType *elem_type = mono_reflection_type_get_type(refltype);
+        MonoClass *mono_class = mono_class_from_mono_type(elem_type);
+        GDMonoClass *klass = GDMono::get_singleton()->get_class(mono_class);
+
+        if (klass == GDMonoUtils::get_class_native_base(klass)) {
+            // If we're trying to get native objects, just check the inheritance list
+            StringName native_class_name = GDMonoUtils::get_native_godot_class_name(klass);
+            for (int i = 0; i < nodes.size(); ++i)
+                if (ClassDB::is_parent_class(nodes[i]->get_class(), native_class_name))
+                    ret.push_back(nodes[i]);
```
Comment (neikeq) [inline modules/mono/glue/scene_tree_glue.cpp:58]: Code style requires braces on single-statement `if`/`else` and `for` blocks.

File: modules/mono/glue/scene_tree_glue.h [hunk 6]
```diff
@@ -0,0 +1,50 @@
+/*************************************************************************/
+/*  collections_glue.h                                                   */
```
Comment (neikeq) [inline modules/mono/glue/scene_tree_glue.h:2]:
```suggestion
/*  scene_tree_glue.h                                                   */
```

**Review by neikeq — CHANGES_REQUESTED** [review]
Implementation looks good, but some style changes are required. Also clang-format is making CI fail because of wrong formatting.

**Comment by paulloz** [comment]
Sorry 🙏,  
Everything should be good now.

It's a new setup and my clang-format wasn't exactly telling me the same things as the CI so I ended up running multiple Travis builds 😭. I'll configure that more properly on my machine for the next one.

**Review by neikeq — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**